uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -106,19 +129,20 @@ def on_task_complete(self, task):
 
     def handle_connect(self, event):
         self.connect_event = event
-        request = event.h11request
-        headers = [(key.lower(), value) for key, value in request.headers]
-        path, _, query_string = request.target.partition(b"?")
+        headers = [(key.lower(), value) for key, value in event.extra_headers]
+        # todo: Hypercorn adds (b"host", event.host.encode()), should Uvicorn do too?
```

Comment:
Yes looks like it to me. Had a quick look at the `wsproto` docs, and extra_headers + host seems to make sense here.

Reply:
Let's add that and drop the comment.